aries_vcx/src/handlers/connection/inviter/state_machine.rs

```diff
@@ -493,15 +447,14 @@ impl SmConnectionInviter {
             .set_service_endpoint(new_service_endpoint)
             .set_keys(new_recipient_keys, new_routing_keys)
             .ask_for_ack()
-            .set_thread_id(&request.id.0)
+            .set_thread_id(&request.get_thread_id().ok_or(VcxError::from_msg(VcxErrorKind::InvalidJson, "Missing ~thread decorator field in request"))?)
```

Comment:
I don't quite remember - can Connection Request have non-set thread?

Reply:
It is marked as TODO to make it mandatory.